src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -1120,7 +1120,7 @@ public int hash(DynamicObject string,
 
         @Specialization
         public DynamicObject initializeJavaString(DynamicObject self, String from) {
-            StringOperations.setRope(self, StringOperations.encodeRope(from, ASCIIEncoding.INSTANCE));
+            StringOperations.setRope(self, StringOperations.encodeRope(from, UTF8Encoding.INSTANCE));
```

Comment:
UTF-8 is safer for sure. But I think this starts getting into questions like is a j.l.String an "external" string? Should it be subject to the external encoding then?

Reply:
a j.l.String which comes from interop could be from anywhere (e.g. a literal String in JS code). I think j.l.String are meant to be textual and non-binary in most cases. So I think UTF-8 is a good choice here for compatibility with other Ruby Strings.

Reply:
That doesn't work for Strings coming from C which should be tagged as binary, so String.new(javaString) now requires an Encoding to be passed.

Reply:
Right. I think we want to delineate between strings from other languages and strings from NFI/Sulong. We haven't done such a good job of this before, but we should be using `Encoding.default_external` when making system calls in most (all?) cases.

Reply:
For info, is `Encoding.default_external` the same as `Encoding.find('locale')` ?